pq-crypto/EVP_kem.c

```diff
@@ -0,0 +1,95 @@
+// -----------------------------------------------------------------------------
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// KEM Library
+// Abstract: Post-Quantum Algorithm API
+// -----------------------------------------------------------------------------
+
+#include <stdint.h>
+#include <stdlib.h>
+#include "EVP_kem.h"
+#include "internal.h"
+
+struct pq_kem evp_sike_p434_r3 = {
+        .name = "SIKEp434r3-KEM",
+        .public_key_length = SIKE_P434_R3_PUBLIC_KEY_BYTES,
+        .private_key_length = SIKE_P434_R3_PRIVATE_KEY_BYTES,
+        .shared_secret_key_length = SIKE_P434_R3_SHARED_SECRET_BYTES,
+        .ciphertext_length = SIKE_P434_R3_CIPHERTEXT_BYTES,
+        .generate_keypair = &sike_p434_r3_crypto_kem_keypair,
+        .encapsulate = &sike_p434_r3_crypto_kem_enc,
+        .decapsulate = &sike_p434_r3_crypto_kem_dec,
+};
+
+int pq_kem_params_alloc(pq_kem kem, pq_kem_params *key_params) {
+    key_params->kem = &evp_sike_p434_r3;
```

Comment:
1. To generalize this, I don't think we want to hardcode `evp_sike_p434_r3` here. You added `pq_kem kem` as an argument, so you're halfway there :)

```
int pq_kem_params_alloc(pq_kem *kem, pq_kem_params *kem_params) {
    kem_params->kem = kem;
    ...
```

2. I like the previous name you had for the second argument, `kem_params` instead of `key_params`.